onnxruntime/core/providers/rknpu/onnx_converter.h

```diff
@@ -0,0 +1,239 @@
+// Copyright 2020 rock-chips.com Inc.
+
+#pragma once
+
+#include <onnx/onnx_pb.h>
+
+#include <map>
+#include <memory>
+#include <utility>
+#include <vector>
+#include <unordered_map>
+#include <string>
+
+#include "shaper.h"
+
+#include "rknpu/rknpu_pub.h"
+
+namespace onnxruntime {
+namespace rknpu {
+class OnnxConverter {
+ private:
+  Shaper shaper_;
+
+  enum class FuseCode { FUSED_NONE,
+                        FUSED_RELU,
+                        FUSED_RELU1,
+                        FUSED_RELU6 };
+
+  std::map<std::string, std::string> name_map_;
+
+  ONNX_NAMESPACE::ModelProto model_proto_;
+  rk::nn::Graph* graph_;
+  std::vector<int> skipped_act_;
+  std::vector<std::string> dequantize_after_;
+
+  std::vector<std::string> operands_;
+  std::map<std::string, std::shared_ptr<rk::nn::Tensor>> rk_tensors_;
+
+  // for GetSupportedNodes
+  std::map<std::string, std::vector<uint32_t>> tensor_dims_;
+
+  std::vector<void*> free_list_;  // remember free
+
+  std::pair<std::pair<int, ONNX_NAMESPACE::NodeProto>, FuseCode>
+  FindActivation(const ONNX_NAMESPACE::ModelProto& model_proto,
+                 const std::string& output_name);
+
+  std::shared_ptr<rk::nn::Tensor>
```

Comment:
>std::shared_ptr<rk::nn::Tensor> [](start = 2, length = 31)

What is the purpose of the shared_ptr?

Reply:
std::shared_ptr<rk::nn::Tensor> is used by many interfaces of rknpu_ddk, is there any unreasonable for this?